addons/wowl/static/src/web_enterprise/navbar/home_menu.ts

```diff
@@ -0,0 +1,598 @@
+//                     optional: 1,
+//                     shape: {
+//                         iconClass: String,
+//                         color: String,
+//                         backgroundColor: String,
+//                     },
+//                 }],
+//                 webIconData: { type: String, optional: 1 },
+//                 xmlid: String,
+//             },
+//         },
+//     },
+//     menuItems: {
+//         type: Array,
+//         element: {
+//             type: Object,
+//             shape: {
+//                 action: String,
+//                 id: Number,
+//                 label: String,
+//                 menu_id: Number,
+//                 parents: String,
+//                 webIcon: Boolean,
+//                 xmlid: String,
+//             },
+//         },
+//     },
+//};
+HomeMenu.template = "HomeMenu";
+
+interface HomeMenuService {
+  hasHomeMenu: boolean;
+  toggle(fswitch?: boolean): void;
+}
+
+const homeMenuSymbol = Symbol("homeMenu");
+export function makeHomeMenuService(): Service<HomeMenuService> {
+  return {
+    name: "home_menu",
+    dependencies: ["action_manager", "router"],
+    deploy(env: OdooEnv) {
+      let hasHomeMenu: boolean = false;
+      let previousHash: any;
+      const bus = new EventBus();
+
+      class HMWrap extends Component {
+        static components = { HomeMenu };
+        static template = tags.xml`
+        <t>
+          <HomeMenu t-if="hm.hasHomeMenu" />
+          <div t-else=""></div>
+        </t>`;
+
+        hm = useHomeMenuToggle(() => this.render());
+      }
+      odoo.mainComponentRegistry.add("HomeMenu", HMWrap);
+
+      function callOnToggled() {
+        bus.trigger("HOME-MENU:TOGGLED");
+        if (hasHomeMenu) {
+          previousHash = env.services.router.current.hash;
+          env.services.router.pushState({ home: "" }, true);
+          env.bus.trigger("ACTION_MANAGER:UI-UPDATED", "home");
+        } else if (previousHash) {
+          env.services.router.pushState(previousHash, true);
+          previousHash = null;
+        }
+        adaptDom();
+      }
+
+      async function toggle(fswitch?: boolean): Promise<void> {
+        if (fswitch === undefined) {
+          fswitch = !hasHomeMenu;
+        }
+        if (fswitch) {
+          await clearUncommittedChanges(env);
+        }
+        if (fswitch !== hasHomeMenu) {
+          hasHomeMenu = fswitch;
+          callOnToggled();
+        }
+      }
+
+      function getActionManagerEl() {
+        return document.body.querySelector('.o_action_manager');
+      }
+
+      function adaptDom() {
+        const amEl = getActionManagerEl();
+        if (!amEl) {
+          return;
+        }
+        if (hasHomeMenu) {
+          amEl.classList.add('o_hidden');
+        } else {
+          amEl.classList.remove('o_hidden');
+        }
+      }
+
+      env.bus.on("ACTION_MANAGER:UI-UPDATED", null, (mode) => {
+        if (mode !== "new" && mode !== "home") {
```

Comment:
in practice this means that we need one more nextAnimFrame  to have the HomeMenu destroyed when a legit controller is put in the DOM (a bit like navbar...)

Reply:
as soon as it doesn't flicker it's fine